File: src/main/java/com/github/liebharc/queryenrichment/EqualityFilter.java
package com.github.liebharc.queryenrichment;

import java.util.Objects;

class EqualityFilter<TAttribute, TParameter> extends FilterStep<TAttribute, TParameter> {
    private static final long serialVersionUID = 6490446172472673292L;

    EqualityFilter(ExecutableStep<TAttribute, TParameter> innerStep, SimpleExpression expression) {
        super(innerStep, expression);
    }

    @Override
    public void enrich(IntermediateResult result, TParameter parameter) {
        innerStep.enrich(result, parameter);
        final TAttribute value = result.get(this.getAttribute());

        if (!Objects.equals(value, expression.getValue())) {
            result.stopProcessing();
        }
    }

    @Override
    public String toString() {
        return "EqualityFilter{" +
                "innerStep=" + innerStep +
                ", expression=" + expression +
                '}';
    }
}

File: src/main/java/com/github/liebharc/queryenrichment/EqualityFilter.kt
package com.github.liebharc.queryenrichment

internal class EqualityFilter<TAttribute, TParameter>(innerStep: ExecutableStep<TAttribute, TParameter>, expression: SimpleExpression<TAttribute>) : FilterStep<TAttribute, TParameter>(innerStep, expression) {

    override fun column(queryInformation: QueryInformation): String? = innerStep.column(queryInformation)

    override fun enrich(result: IntermediateResult, parameter: TParameter) {
        innerStep.enrich(result, parameter)
        val value = result[this.attribute]

        if (value != expression.value) {
            result.stopProcessing()
        }
    }

    override fun toString(): String {
        return "EqualityFilter{" +
                "innerStep=" + innerStep +
                ", expression=" + expression +
                '}'.toString()
    }

    companion object {
        private const val serialVersionUID = 6490446172472673292L
    }
}
